nautobot/extras/models/relationships.py

```diff
@@ -533,6 +558,144 @@ def clean(self):
                     " are present in the database, delete all associations first before modifying the destination type."
                 )
 
+    def skip_required(self, referenced_instance_or_class, side):
+        """
+        This takes an instance or class and a side and checks if it should
+        be skipped or not when validating required relationships.
+        It will skip when any of the following conditions are True:
+         - a relationship is marked as symmetric
+         - a relationship references itself
+         - if a required model class is None (if it doesn't exist yet -- unimplemented/uninstalled plugins for instance)
+
+        Args:
+            referenced_instance_or_class: model instance or class
+            side: side of the relationship being checked
+
+        Returns: Bool
+        """
+
+        # Not enforcing required symmetric relationships
+        if self.symmetric:
+            return True
+
+        # Skip self-referencing
+        if ContentType.objects.get_for_model(referenced_instance_or_class) == getattr(self, f"{side}_type"):
+            return True
+
+        required_model_class = getattr(self, f"{RelationshipSideChoices.OPPOSITE[side]}_type").model_class()
+        # Handle the case where required_model_class is None (e.g., relationship to a plugin
+        # model for a plugin that's not installed at present):
+        if required_model_class is None:
+            return True
+
+        return False
+
+    @classmethod
+    def required_related_objects_errors(
+        cls, instance_or_class, output_for="ui", initial_data=None, relationships_key_specified=False
+    ):
+        """
+        Args:
+            instance_or_class: either a model instance or class name to check its required relationships
+            output_for: either "ui" or "api" depending on usage
+            initial_data: submitted form/serializer data to validate against
+            relationships_key_specified: if the "relationships" key was provided or not
+        Returns:
+            List of field error dicts if any are found
+        """
+
+        required_relationships = Relationship.objects.get_required_for_model(instance_or_class)
+        relationships_field_errors = []
+        for relation in required_relationships:
+
+            opposite_side = RelationshipSideChoices.OPPOSITE[relation.required_on]
+
+            if relation.skip_required(instance_or_class, opposite_side):
+                continue
+
+            if relation.has_many(opposite_side):
+                num_required_verbose = "at least one"
+            else:
+                num_required_verbose = "a"
+
+            if output_for == "api":
+                # If this is a model instance and the relationships json data key is missing, check to see if
+                # required relationship associations already exist, and continue (ignore validation) if so
+                if (
+                    instance_or_class.present_in_database is True
```

Comment:
`present_in_database` is only an attribute of a model instance, not a model class. Is `instance_or_class` a misnomer, is this parameter always going to be an instance?